dispatcher/brokers/pg_notify.py

```diff
@@ -18,11 +18,17 @@
 
 async def acreate_connection(**config) -> psycopg.AsyncConnection:
     "Create a new asyncio connection"
-    return await psycopg.AsyncConnection.connect(**config)
+    connection = await psycopg.AsyncConnection.connect(**config)
+    if not connection.autocommit:
+        await connection.set_autocommit(True)
+    return connection
 
 
 def create_connection(**config) -> psycopg.Connection:
-    return psycopg.Connection.connect(**config)
+    connection = psycopg.Connection.connect(**config)
+    if not connection.autocommit:
+        connection.set_autocommit(True)
```

Comment:
For future readers, this _has_ a hazard, which is that sometimes you _want_ to send messages as a part of a transaction, for transactional integrity. This is documented - messages will not send if transactions are rolled back, and they will delay until the commit. This feature is actually used in AWX.

But in any case someone relies on that behavior, they need to use the connection factory. In other words, provide their own connection. This is only called if dispatcher, itself, has to create a connection for some reason, and is bypassed by passing a connection factory. Specifically, this is how we would use the Django connection without creating a new connection. _That_ connection may keep autocommit off.